homeassistant/components/digital_ocean/schemas.py

```diff
@@ -0,0 +1,26 @@
+"""Holds validation schemas used across the integration."""
+import voluptuous as vol
+
+from homeassistant.helpers import config_validation as cv
+
+from . import constants as const
+
+UPDATE_DOMAIN_RECORD_SCHEMA = vol.Schema(
+    {
+        vol.Required(const.SERVICE_PARAM_DOMAIN_NAME): cv.matches_regex(
+            r"^(?!-)[A-Za-z0-9-]+([\\-\\.]{1}[a-z0-9]+)*\\.[A-Za-z]{2,6}$"
```

Comment:
Please move the regexes to separate constants with a summary of what they do, e.g:

```py
# blablabla
DOMAIN_NAME_REGEX = r"^(?!-)[A-Za-z0-9-]+([\\-\\.]{1}[a-z0-9]+)*\\.[A-Za-z]{2,6}$"
```

Also, this regex looks weird, are you sure it's correct:
- Why does it have a capturing group?
- What's up with this: `[\\-\\.]{1}`, it's an overly complicated way of saying there should be exactly one `\` or a `.` as I understand it?
- Why do we require a TLD of 2-6 characters, since TLDs are now free for all, isn't that wrong?